language/vm/vm-runtime/src/system_txn/block_metadata_processor.rs

```diff
@@ -0,0 +1,48 @@
+use crate::{
+    code_cache::module_cache::ModuleCache,
+    data_cache::BlockDataCache,
+    process_txn::execute::ExecutedTransaction,
+    txn_executor::{TransactionExecutor, LIBRA_SYSTEM_MODULE},
+};
+use lazy_static::lazy_static;
+use libra_types::{
+    block_metadata::BlockMetadata,
+    identifier::Identifier,
+    transaction::TransactionOutput,
+    vm_error::{StatusCode, VMStatus},
+};
+use vm::transaction_metadata::TransactionMetadata;
+use vm_runtime_types::value::Value;
+
+lazy_static! {
+    static ref BLOCK_PROLOGUE: Identifier = Identifier::new("block_prologue").unwrap();
+}
+
+pub(crate) fn process_block_metadata<'alloc, P>(
+    block_metadata: BlockMetadata,
+    module_cache: P,
+    data_cache: &mut BlockDataCache<'_>,
+) -> TransactionOutput
+where
+    P: ModuleCache<'alloc>,
+{
+    // TODO: How should we setup the metadata here?
+    let txn_data = TransactionMetadata::default();
```

Comment:
I am assuming this is not really needed by the Move code right now right?
and we will change things around this so it seems all good to me now...

Reply:
I think we should resolve this (or at least part of it) before landing; it feels like a dangerous hole to leave open. We should at least change this to make the gas limit u64_MAX (it's some arbitrarily chosen value like `10k` in `default()` to avoid confusing out-of-gas errors if the block prologue becomes more expensive to execute in the future.